lib/compat/wordpress-6.2/class-gutenberg-rest-block-patterns-controller-6-2.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/**
+ * REST API: Gutenberg_REST_Block_Patterns_Controller_6_2 class
+ *
+ * @package    Gutenberg
+ * @subpackage REST_API
+ */
+
+/**
+ * Core class used to access block patterns via the REST API.
+ *
+ * @since 6.0.0
+ *
+ * @see WP_REST_Controller
+ */
+class Gutenberg_REST_Block_Patterns_Controller_6_2 extends Gutenberg_REST_Block_Patterns_Controller_6_1 {
+	/**
+	 * Defines whether remote patterns should be loaded.
+	 *
+	 * @since 6.0.0
+	 * @var bool
+	 */
+	private $remote_patterns_loaded;
+
+	/**
+	 * An array that maps old categories names to new ones.
+	 *
+	 * @since 6.2.0
+	 * @var array
+	 */
+	protected static $categories_migration = array(
+		'buttons' => 'call-to-action',
+		'columns' => 'text',
+		'query'   => 'posts',
+	);
+
+	/**
+	 * Registers the routes for the objects of the controller.
+	 *
+	 * @since 6.0.0
+	 */
+	public function register_routes() {
```

Comment:
Is this definition still needed? It doesn't seem to differ from `WP_REST_Block_Patterns_Controller`.

Reply:
Yes, because of this line: `'callback'            => array( $this, 'get_items' ),` We're updating `get_items` here and it would call `Gutenberg_REST_Block_Patterns_Controller_6_1`' one otherwise.